packages/web3-types/src/json_rpc_types.ts

```diff
@@ -55,8 +55,9 @@ export interface JsonRpcNotification<T = JsonRpcResult> {
 	readonly jsonrpc: JsonRpcIdentifier;
 	readonly method: string; // for subscription
 	readonly params: SubscriptionParams<T>; // for subscription results
-	readonly result: never;
+	readonly result?: never;
```

Comment:
Needed to add this otherwise we can't instantiate JsonRpcNotification

Reply:
this will be breaking change & , the result property is marked as a required property of type never. This means that any object conforming to this interface must have a result property, but it can never be assigned a value because the never type represents a state that should never occur, so its a contradictory definition.

Reply:
This is a bug,
Within [processSubscriptionData](https://github.com/web3/web3.js/blob/eab5a68eb5e97864a6b3fba7c3763ec7a2d09bb9/packages/web3-core/src/web3_subscriptions.ts#L107) data must be 
```
data:
			| JsonRpcSubscriptionResult
			| JsonRpcSubscriptionResultOld<Log>
			| JsonRpcNotification<Log>,
```
and within type `JsonRpcSubscriptionResult` result is a string,
we run into an issue in data is `JsonRpcNotification<Log>` when result is never, saying result is missing
```
 test/unit/web3_subscription.test.ts:251:10 - error TS2741: Property 'result' is missing in type '{ jsonrpc: "2.0"; params: { subscription: string; result: { id: string; }; }; method: string; }' but required in type 'JsonRpcNotification<Log>'.

    251    const responseWithNotification: JsonRpcNotification<Log> = {
                 ~~~~~~~~~~~~~~~~~~~~~~~~

      ../web3-types/lib/commonjs/json_rpc_types.d.ts:36:14
        36     readonly result: never;
                        ~~~~~~
        'result' is declared here.
```
Because result mandatory in `JsonRpcSubscriptionResult` 
and never in `responseWithNotification` using `result ?: never` will still enforce the type and allow processSubscriptionData to pass in type responseWithNotification

We do this already in other types already
https://github.com/web3/web3.js/blob/eab5a68eb5e97864a6b3fba7c3763ec7a2d09bb9/packages/web3-types/src/json_rpc_types.ts#L31